etcdserver/api/etcdhttp/peer.go

```diff
@@ -79,3 +105,55 @@ func (h *peerMembersHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+func (h *peerMemberPromoteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if !allowMethod(w, r, "POST") {
+		return
```

Comment:
`http.Error` here before return?

Reply:
Function name is misleading. The function actually writes error before returning false.